File: Derelict2D-1.0/src/br/odb/derelict2d/ShowOutcomeActivity.java
package br.odb.derelict2d;

import br.odb.gamelib.android.GameView;
import br.odb.gamerendering.rendering.AssetManager;
import br.odb.gamerendering.rendering.DisplayList;
import br.odb.gamerendering.rendering.RenderingNode;
import br.odb.gamerendering.rendering.SVGRenderingNode;
import br.odb.libsvg.SVGGraphic;
import br.odb.utils.Rect;
import br.odb.utils.math.Vec2;
import android.app.Activity;
import android.media.MediaPlayer;
import android.os.Bundle;
import android.widget.TextView;

public class ShowOutcomeActivity extends Activity {

	MediaPlayer theme = null;
	
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_show_outcome);
		
		( ( TextView ) findViewById( R.id.txtOutcome ) ).setText( getIntent().getExtras().getString( "outcome" ) );
		
		if (theme == null
				&& ((Derelict2DApplication) getApplication()).mayEnableSound()) {
			theme = MediaPlayer.create(this, R.raw.derelicttheme);
			theme.setLooping(true);
			theme.start();
		}		
	}
	
	@Override
	protected void onResume() {
		super.onResume();
		
		DisplayList dl = new DisplayList("dl");
		AssetManager resManager = ((Derelict2DApplication) getApplication())
				.getAssetManager();

		SVGGraphic graphic = resManager.getGraphics("logo");

		float scale = 1;
		Vec2 trans = new Vec2();

		GameView gvSplash = (GameView) findViewById( R.id.gvGameLogoOutcome );
		
		if (gvSplash.getWidth() > 0 && gvSplash.getHeight() > 0) {

			Rect bound = graphic.makeBounds();

			// não me interessa a parte acima da "página".
			float newWidth = bound.p1.x;
			float newHeight = bound.p1.y;

			if (newWidth > newHeight) {
				scale = gvSplash.getWidth() / newWidth;
				trans.y = (gvSplash.getHeight() - (bound.p1.y * scale)) / 2.0f;
			} else {
				scale = gvSplash.getHeight() / newHeight;
				trans.x = (gvSplash.getWidth() - (bound.p1.x * scale)) / 2.0f;
			}
		}

		graphic = graphic.scale(scale, scale);
		SVGRenderingNode node = new SVGRenderingNode(graphic, "title");
		node.translate.set(trans);

		
		dl .setItems(new RenderingNode[] { node });

		gvSplash.setRenderingContent(dl);
		gvSplash.postInvalidate();		
				
	}
	
	@Override
	protected void onPause() {
		if (theme != null) {

			theme.stop();
		}
		super.onPause();
	}

	@Override
	protected void onDestroy() {
		if (theme != null) {

			theme.stop();
		}
		super.onDestroy();
	}
}

File: Derelict2D-trunk/app/src/main/java/br/odb/derelict2d/ShowOutcomeActivity.kt
package br.odb.derelict2d

import android.app.Activity
import android.media.MediaPlayer
import android.os.Bundle
import br.odb.gamerendering.rendering.DisplayList
import br.odb.gamerendering.rendering.RenderingNode
import br.odb.gamerendering.rendering.SVGRenderingNode
import br.odb.gameutils.math.Vec2
import kotlinx.android.synthetic.main.activity_show_outcome.*

class ShowOutcomeActivity : Activity() {

    private var theme: MediaPlayer? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_show_outcome)

        txtOutcome.text = intent.extras!!.getString("outcome")

        if (theme == null
                && (application as Derelict2DApplication).mayEnableSound()) {

            theme = MediaPlayer.create(this, R.raw.derelicttheme)

            if ( theme != null ) {
                theme!!.isLooping = true
                theme!!.start()
            }
        }
    }

    override fun onResume() {
        super.onResume()
        val dl = DisplayList("dl")

        val resManager = (application as Derelict2DApplication)
                .assetManager

        var graphic = resManager.getGraphics("logo")
        var scale = 1f
        val trans = Vec2()
        val gvSplash = gvGameLogoOutcome

        if (gvSplash.width > 0 && gvSplash.height > 0) {
            val bound = graphic.makeBounds()
            // não me interessa a parte acima da "página".
            val newWidth = bound.p1.x
            val newHeight = bound.p1.y
            if (newWidth > newHeight) {
                scale = gvSplash.width / newWidth
                trans.y = (gvSplash.height - bound.p1.y * scale) / 2.0f
            } else {
                scale = gvSplash.height / newHeight
                trans.x = (gvSplash.width - bound.p1.x * scale) / 2.0f
            }
        }
        graphic = graphic.scale(scale, scale)
        val node = SVGRenderingNode(graphic, "title")
        node.translate.set(trans)
        dl.items = arrayOf<RenderingNode>(node)
        gvSplash.setRenderingContent(dl)
        gvSplash.postInvalidate()
    }

    override fun onPause() {
        if (theme != null) {
            theme!!.stop()
        }
        super.onPause()
    }

    override fun onDestroy() {
        if (theme != null) {
            theme!!.stop()
        }
        super.onDestroy()
    }
}